internal/llm/tools/shell/shell.go

```diff
@@ -47,7 +47,7 @@ func GetPersistentShell(workingDir string) *PersistentShell {
 		shellInstance = newPersistentShell(workingDir)
 	})
 
-	if !shellInstance.isAlive {
+	if shellInstance == nil || !shellInstance.isAlive {
```

Comment:
```suggestion
	if shellInstance != nil && !shellInstance.isAlive {
```
Is this what we should do instead ?

Reply:
No, we want to create a shell instance when it doesn't exist or when its not active

Reply:
Ah ok, but we need to change the next line also, we are trying to get .cwd from shellInstance and that shellInstance could be nil in this case.

Reply:
Funny thing is I was getting that nil error, but adding that nil check did do the fix. Not sure why it didn't panic on the next line 😆

Will find out the reason and modify it if needed

Reply:
@kujtimiihoxha I've updated the line to use the provided workingDir